### DISCUSSION TITLE
TerrainLayer + Cartesian Coordinate GeoJson

### DISCUSSION THREAD

**Discussion by warflash** [visgl/deck.gl#6431] [Q&A] [answered]
Hello!

I am currently working on a 3D version of a games minimap and this library seems like it would be a nice fit for that.
Figuring out the best combination of components has been challenging the past few days though, so I was wondering if anyone has advice on that usecase.
The game uses a cartesian coordinate system for placing markers, so I somehow need to figure out how to combine a TerrainLayer with a cartesian coordinate system.

---
Edit/Update: 
Seems to me like using geojson + TerrainLayer + MapView is the play here.
What's left now is that the cartesian coords given to the geojson layer dont appear where they're supposed to be, which I assume is due to TerrainLayer using lat/lng. 
Is there a way to convert between the two natively (eg. by telling deck what the overall terrain bounds are, similar to how its done in leaflet) . Or should I go down the route of using a modelMatrix to force my coords into fitting properly, similar to how it's described here: https://deck.gl/docs/developer-guide/coordinate-systems#transforming-positions?

Am very grateful for any tips on what direction I should take or even examples if someone had a similar setup before, thanks!

**Comment by Pessimistress** [comment, marked as answer]
You are correct that TerrainLayer only supports Web Mercator projection. If you open a bug report we can see to getting it fixed.
